web/views.py

```
@@ -906,15 +906,26 @@ def add_review(request, slug):
     if request.user != course.teacher:
-        messages.error(request, "Only the course teacher can delete the course!")
+        messages.error(request, "You are not authorized to delete this course.")
         return redirect("course_detail", slug=slug)
 
     if request.method == "POST":
+        # Delete the course image if it exists
+        if course.image and os.path.isfile(course.image.path):
+            try:
+                os.remove(course.image.path)
+            except (PermissionError, OSError) as e:
+                logger.warning(f"Failed to delete image for course {course.slug}: {e}")
```

Ruff (0.8.2)

Logging statement uses f-string

(G004)